### PR TITLE
todo list add & remove

### PR DESCRIPTION
**Pull request by AbuTalha3**
In this PR the requirements were;-
- Adding a new item function to work.
- Removing a selected item to function.
- Marking a selected item as complete.
 -Removing all items marked as complete at once.
- Added features of the Project to README file.

### CODE REVIEW

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @AbuTalha3 👋👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] A function for adding a new task has been implemented👍.
- [x] A function for editing a task has been implemented👍.
- [x] A function for deleting a task has been implemented👍.
- [x] A function to clear completed tasks has been implemented👍.
- [x] Linters are passing👍.


## Required Changes ♻️

- [ ] Nice job so far👍. but just as required for this milestone, kindly remove all hardcoded items from the task array and create a new js file and add all the new functionalities in it😉.
image

image

- [ ] Your changes are currenly not stored in localstorage and it is a requirement in this milestone that all changes should be saved in localstorage😉. Kindly do well to make the necessary adjustments👍. When the changes are made, your page should be able to retain it's content when the page is loaded since it renders the data stored in the local storage😉.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/todos.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      index: todos.length + 1,
+    };
+
+    todos.push(todoItem);
+    saveTodosToLocalStorage();
+
+    textInputField.value = '';
+
+    renderTodoItem(todoItem);
+  });
```
Comment (rotimiazeez) [inline src/todos.js:31]: Please note that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique (i.e. if a task is removed, then added back to the list, the index of that task should equal the total number of tasks
Screenshot from 2023-07-15 00-49-27
).

Comment (rotimiazeez) [reply src/todos.js:31]: As stated by the previous CR, kindly remove the hardcoded item from your index.js.

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi @AbuTalha3 😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlight 
* Good job saving all changes to the To-Do List in local storage. 👍🏽

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: src/todos.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  function refreshIt() {
+    const refreshIcon = document.querySelector('.refresh-it');
+    refreshIcon.addEventListener('click', () => {
+      todos = [];
+      todosContainer.innerHTML = '';
+      localStorage.removeItem('todos');
+    });
+  }
```
Comment (cynthiainga) [inline src/todos.js:150]: - [Optional] I noticed that when I click on the refresh button it removes all the items from the list, the refresh button is supposed to refresh the page and not delete the items. Please kindly fix that 🙏🏾

File: src/todos.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-use-before-define */
```
Comment (cynthiainga) [inline src/todos.js:1]: - [Optional] It's not a good approach to disable linters, fixing linters help us to write a clean, clear, and good codebase, kindly make sure to not disable linters 🙏🏾

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 ✔️ 

Hi @AbuTalha3, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] Implemented all requested changes ✔️ 
- [ ] Correct GitHub flow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**